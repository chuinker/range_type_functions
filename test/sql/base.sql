begin;
\set ECHO none
\i sql/range_type_functions.sql
\i sql/range_type_functions_94.sql

\set ECHO all

set datestyle = 'ISO';

select to_range(4,5,'[]',null::int4range);

select to_range('2015-01-01'::date,'2016-01-1','[)',null::daterange);

select to_range(4,null::int4range);

select to_range('2015-01-01'::date,null::daterange);

select element_range_comp(4,'[10,100]'::int4range);

select element_range_comp(10,'[10,100]'::int4range);

select element_range_comp(110,'[10,100]'::int4range);

select is_singleton('[4,5)'::int4range);

select is_singleton('[4,5]'::int4range);


select range_merge('[4,5]'::int4range,'[9,10]'::int4range);

select express_lower_bound_condition('empty'::int4range);
select express_lower_bound_condition('(,)'::int4range);
select express_lower_bound_condition('(4,5]'::int4range);
select express_lower_bound_condition('(4,5]'::int4range,'y.z');
select express_lower_bound_condition('[4,5]'::int4range,'y.z');
select express_lower_bound_condition('[4,5]'::int4range,format('%I.%I','my schema','my ColuMnaME'));
select express_upper_bound_condition('empty'::int4range);
select express_upper_bound_condition('(,)'::int4range);
select express_upper_bound_condition('[4,5)'::int4range,'y.z');
select express_upper_bound_condition('[4,5]'::int4range,'y.z');
select express_bounds_conditions('empty'::int4range);
select express_bounds_conditions('(,)'::int4range);
select express_bounds_conditions('(4,5]'::int4range);
select express_bounds_conditions('(4,5]'::int4range,'y.z');
select express_bounds_conditions('[4,5]'::int4range,'y.z');
select express_bounds_conditions('[4,5)'::int4range,'y.z');
select express_bounds_conditions('[4,5]'::int4range,'y.z');







