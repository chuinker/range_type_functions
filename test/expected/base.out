begin;
\set ECHO none
set datestyle = 'ISO';
select to_range(4,5,'[]',null::int4range);
 to_range 
----------
 [4,6)
(1 row)

select to_range('2015-01-01'::date,'2016-01-1','[)',null::daterange);
        to_range         
-------------------------
 [2015-01-01,2016-01-01)
(1 row)

select to_range(4,null::int4range);
 to_range 
----------
 [4,5)
(1 row)

select to_range('2015-01-01'::date,null::daterange);
        to_range         
-------------------------
 [2015-01-01,2015-01-02)
(1 row)

select element_range_comp(4,'[10,100]'::int4range);
 element_range_comp 
--------------------
                 -1
(1 row)

select element_range_comp(10,'[10,100]'::int4range);
 element_range_comp 
--------------------
                  0
(1 row)

select element_range_comp(110,'[10,100]'::int4range);
 element_range_comp 
--------------------
                  1
(1 row)

select is_singleton('[4,5)'::int4range);
 is_singleton 
--------------
 t
(1 row)

select is_singleton('[4,5]'::int4range);
 is_singleton 
--------------
 f
(1 row)

select range_merge('[4,5]'::int4range,'[9,10]'::int4range);
 range_merge 
-------------
 [4,11)
(1 row)

select express_lower_bound_condition('empty'::int4range);
 express_lower_bound_condition 
-------------------------------
 false
(1 row)

select express_lower_bound_condition('(,)'::int4range);
 express_lower_bound_condition 
-------------------------------
 true
(1 row)

select express_lower_bound_condition('(4,5]'::int4range);
 express_lower_bound_condition 
-------------------------------
 x >= '5'
(1 row)

select express_lower_bound_condition('(4,5]'::int4range,'y.z');
 express_lower_bound_condition 
-------------------------------
 y.z >= '5'
(1 row)

select express_lower_bound_condition('[4,5]'::int4range,'y.z');
 express_lower_bound_condition 
-------------------------------
 y.z >= '4'
(1 row)

select express_lower_bound_condition('[4,5]'::int4range,format('%I.%I','my schema','my ColuMnaME'));
   express_lower_bound_condition   
-----------------------------------
 "my schema"."my ColuMnaME" >= '4'
(1 row)

select express_upper_bound_condition('empty'::int4range);
 express_upper_bound_condition 
-------------------------------
 false
(1 row)

select express_upper_bound_condition('(,)'::int4range);
 express_upper_bound_condition 
-------------------------------
 true
(1 row)

select express_upper_bound_condition('[4,5)'::int4range,'y.z');
 express_upper_bound_condition 
-------------------------------
 y.z < '5'
(1 row)

select express_upper_bound_condition('[4,5]'::int4range,'y.z');
 express_upper_bound_condition 
-------------------------------
 y.z < '6'
(1 row)

select express_bounds_conditions('empty'::int4range);
 express_bounds_conditions 
---------------------------
 false
(1 row)

select express_bounds_conditions('(,)'::int4range);
 express_bounds_conditions 
---------------------------
 true
(1 row)

select express_bounds_conditions('(4,5]'::int4range);
 express_bounds_conditions 
---------------------------
 x >= '5' and x < '6'
(1 row)

select express_bounds_conditions('(4,5]'::int4range,'y.z');
 express_bounds_conditions 
---------------------------
 y.z >= '5' and y.z < '6'
(1 row)

select express_bounds_conditions('[4,5]'::int4range,'y.z');
 express_bounds_conditions 
---------------------------
 y.z >= '4' and y.z < '6'
(1 row)

select express_bounds_conditions('[4,5)'::int4range,'y.z');
 express_bounds_conditions 
---------------------------
 y.z >= '4' and y.z < '5'
(1 row)

select express_bounds_conditions('[4,5]'::int4range,'y.z');
 express_bounds_conditions 
---------------------------
 y.z >= '4' and y.z < '6'
(1 row)

